react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift

```diff
@@ -27,17 +27,22 @@ public class RNReactNativeGutenbergBridge: RCTEventEmitter {
                     callback(nil)
                     return
                 }
-                if (allowMultipleSelection) {
-                    let formattedMedia = media.map { (id, url, type) in
-                        return [mediaDictKeys.IDKey: id, mediaDictKeys.URLKey: url, mediaDictKeys.TypeKey: type]
-                    }
-                    callback([formattedMedia])
+                let mediaToReturn: [MediaInfo]
+                if allowMultipleSelection {
+                    mediaToReturn = media
                 } else {
-                    guard let (mediaID, mediaURL, mediaType) = media.first else {
-                        callback(nil)
-                        return
-                    }
-                    callback([[mediaDictKeys.IDKey: mediaID, mediaDictKeys.URLKey: mediaURL, mediaDictKeys.TypeKey: mediaType]])
+                    mediaToReturn = Array(media.prefix(1))
+                }
+
+                let jsFormattedMedia = mediaToReturn.map { mediaInfo in
+                    return [mediaDictKeys.IDKey: mediaInfo.id as Any,
+                            mediaDictKeys.URLKey: mediaInfo.url as Any,
+                            mediaDictKeys.TypeKey: mediaInfo.type as Any]
+                }
+                if allowMultipleSelection {
+                    callback([jsFormattedMedia])
+                } else {
+                    callback(jsFormattedMedia)
```

Comment:
It might be good to merge this and send the full array. If multi select is false, the callback should bring just one media item. Maybe we can enforce that via the callback signature, and keep the bridge simpler 🤔 

Again, thinking about the future!

Reply:
We don't have control on the callback interface it's defined by RN library.
It's always expecting an array where each item is an argument of the JS function.
What we can do is to cleanup the callback functions on the other side to always expect an array of media.

Reply:
Let's do that! On another PR someday in the future 😬
It might be good to create a ticket with all these little cleanup tasks to keep track of them.